### DISCUSSION TITLE
Virtualisation for big tables

### DISCUSSION THREAD

**Discussion by mossa-Sammer** [iddan/react-spreadsheet#349] [Ideas]
I have a large set of data which is an excel file with thousands of rows, when i read the file and pass the data to the SpreadSheet Component, the UI lags, i searched the docs but didn't find anything related to my problem, help please

**Comment by dpejic** [comment]
Create pagination

**Comment by ghost** [comment]
Hello, is there any plan to have this feature implemented?

**Comment by iddan** [comment]
Hey @tmanuel94, there are some features I would like to tackle before as the library can handle pretty large datasets (due to efficient updating of the DOM). Did you try to use the library with a dataset and it was too big for it?

**Comment by ghost** [comment]
Hi @iddan, I tried with a csv which has 1922 rows * 9 cols and it's just too much to be handled. I don't know if maybe there's something that I can do on my part to have better performance with this kind o f files

**Comment by iddan** [comment]
Have you tried using the React Dev Tools? You should make sure only the components you expect are rendered?

**Comment by ghost** [comment]
@iddan I'm not sure on how to implement it as of now, it's a new topic for me. I'll try to look into that! As a tip, do you think that something like an IntersectionObserver could be the way to go?

**Comment by iddan** [comment]
Not sure. My tip is to use `useMemo` and `useCallback` to make sure React is not re-rendering the spreadsheet when not needed.
